src/parsers/slice/grammar.rs

```diff
@@ -507,6 +507,18 @@ fn construct_attribute(
     parser.ast.add_element(OwnedPtr::new(attribute))
 }
 
+fn unescape_string_literal(s: &str) -> String {
+    let mut is_escaped = false;
+    s.chars()
+        .filter(|c| {
+            // If `c` is a backslash that isn't escaped, set that the next character is escaped,
+            // and return false to remove the backslash from the string.
+            is_escaped = *c == '\\' && !is_escaped;
```

Comment:
```suggestion
            let is_backslash = *c == '\\';
            is_escaped = is_backslash && !is_escaped;
```

I found this filter required a bit of starting at to verify the logic. I think it could be a little easier to read if we extract `*c == '\\'` to its own variable.